include/rmm/device_buffer.hpp

```diff
@@ -410,4 +392,22 @@ class device_buffer {
       mr::get_default_resource()};  ///< The memory resource used to
                                     ///< allocate/deallocate device memory
 };
+
+/**--------------------------------------------------------------------------*
+ * @brief Copies rmm::device_buffer to a preallocated host buffer.
+ *-------------------------------------------------------------------------**/
+void copy_to_host(device_buffer& db, void* hb, cudaStream_t stream = 0) {
+  if (hb == nullptr) {
+    throw std::runtime_error{"Cannot copy to `nullptr`."};
+  }
+  db.set_stream(stream);
```

Comment:
I think now that this is a free function you should not set the stream on the source device_buffer.  Just use the specified stream directly in the cudaMemcpyAsync.

Reply:
Good point. Thanks. Changed.